src/TranslationFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend;
+
+use Symfony\Component\Translation\Loader\JsonFileLoader;
+use Symfony\Component\Translation\Translator;
+use Symfony\Component\Translation\MessageSelector;
+
+/**
+ * @license GNU GPL v2+
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
+ */
+class TranslationFactory {
```

Comment:
I don't see a real issue with making the main factory bigger. Splitting it forces you to make more things public, and I think it often makes things harder to find rather than easier.

Such tests can be nice though.